src/compiler/transpile/datacollection/component-decorator.ts

```diff
@@ -47,14 +46,24 @@ It will be removed in future versions. Please use the "hostData()" method instea
     assetsDirsMeta: [],
     hostMeta: getHostMeta(diagnostics, componentOptions.host),
     dependencies: [],
-    jsdoc: serializeSymbol(checker, symbol)
+    jsdoc: serializeSymbol(checker, symbol)    
   };
 
-  // normalizeEncapsulation
-  cmpMeta.encapsulationMeta =
-      componentOptions.shadow ? ENCAPSULATION.ShadowDom :
+  // shadow dom options
+  if(componentOptions.shadow){
+    cmpMeta.encapsulationMeta = ENCAPSULATION.ShadowDom
+    if(isShadowDomOptions(componentOptions.shadow)) {
```

Comment:
```suggestion
    if(isShadowDomOptions(componentOptions.shadow)) {
        cmpMeta.shadowRootInit = componentOptions.shadow;
```

Reply:
remove the DEFAULT_SHADOW_ROOT_INIT from here, it should be provided by the runtime automatically